### PR TITLE
✨ add aws.eks.nodegroup resource (on aws.eks.cluster)

### PR DESCRIPTION
**Pull request by vjeffrey**
Screenshot 2024-05-27 at 20 02 01

### CODE REVIEW

File: providers/aws/resources/aws.lr [hunk 1]
```diff
@@ -2914,6 +2954,35 @@ aws.eks {
   clusters() []aws.eks.cluster
 }
 
+private aws.eks.nodegroup {
```
Comment (tas50) [inline providers/aws/resources/aws.lr]: nodegroup should go in the expect.txt file

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 2]
```diff
@@ -1502,6 +1506,27 @@ resources:
...
+      instanceTypes: {}
+      labels: {}
+      name: {}
+      nodeRole: {}
+      region: {}
+      scalingConfig: {}
+      status: {}
+      tags: {}
+    is_private: true
+    min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 3]
```diff
@@ -1491,6 +1493,8 @@ resources:
       logging: {}
       name: {}
       networkConfig: {}
+      nodeGroups:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 4]
```diff
@@ -1475,6 +1475,8 @@ resources:
         Use the `aws.eks` resource to assess the configuration of Amazon Elastic Kubernetes Service (EKS).
     fields:
       clusters: {}
+      nodeGroups:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

**Comment by tas50** [comment]
I have a cluster in 177043759486 you can test against

```
cnquery> aws.eks.clusters.first
aws.eks.clusters.first: aws.eks.cluster arn="arn:aws:eks:us-west-2:177043759486:cluster/tim_test" version="1.29" status="ACTIVE"
cnquery> aws.eks.clusters.first.nodeGroups
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 3765ffef-27f5-4c9a-826b-67a70ee548cb, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 002f323b-be8e-479b-a422-809efc5bba81, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 272aa5e9-2d30-4e83-885c-21353412c270, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 7bf26683-b3c9-46b4-989b-496d9a73b786, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 97a45a9e-85c0-4bcc-b83c-032becea5c5a, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: d6fe97cc-e411-4dda-b953-c0248d5f540f, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: acbee903-5b31-4646-91ad-b008b321d039, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 54bd8484-049c-49fa-b306-5a2812d6ce57, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 4b048b2f-d33a-4f72-b99b-dc1b9a4d1682, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 897e2ba1-3292-4a32-8ff9-10e243e33f6a, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: d2a4ab23-cb31-4862-ae17-b4dd34b6fb20, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: f18be530-a56b-4043-a4d6-37df160c3fc2, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 4b54592d-7d58-4e67-8706-44ff26207fbb, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: e3bad53d-34e4-4f72-a453-4d2daf3c8278, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 1220bb1f-bc6a-473f-9c60-5411385facaf, ResourceNotFoundException: No cluster found for name: tim_test.
operation error EKS: ListNodegroups, https response error StatusCode: 404, RequestID: 9f1d4914-7eba-4759-8225-2a4ebe925dd0, ResourceNotFoundException: No cluster found for name: tim_test.
aws.eks.clusters.first.nodeGroups: null
```

**Comment by tas50** [comment]
Poking around it looks like it fails because the cost goes through every region looking for that cluster. The 404s cause a failure. I hardcoded the `getNodegroups` function to just look in `us-west-2` where my cluster is and query returns

```
  nodeGroups: [
    0: aws.eks.nodegroup
  ]
```

File: providers/aws/resources/aws.lr (hunk 1 above)
Comment (tas50) [inline providers/aws/resources/aws.lr]:
```suggestion
private aws.eks.nodegroup @defaults("name scalingConfig.DesiredSize diskSize status") {
```

File: providers/aws/resources/aws_eks.go [hunk 5]
```diff
@@ -123,3 +128,206 @@ func (a *mqlAwsEks) getClusters(conn *connection.AwsConnection) []*jobpool.Job {
...
+
+func (a *mqlAwsEksNodegroup) scalingConfig() (map[string]interface{}, error) {
+	ng, err := a.fetchDetails()
+	if err != nil {
+		return nil, err
+	}
+	return convert.JsonToDict(ng.ScalingConfig)
+}
+
+func (a *mqlAwsEksNodegroup) instanceTypes() (map[string]interface{}, error) {
```
Comment (tas50) [inline providers/aws/resources/aws_eks.go]: This type is failing

```
  instanceTypes: json: cannot unmarshal array into Go value of type map[string]interface {}
```

File: providers/aws/resources/aws_eks.go [hunk 6]
```diff
@@ -123,3 +128,206 @@ func (a *mqlAwsEks) getClusters(conn *connection.AwsConnection) []*jobpool.Job {
...
+		return nil, err
+	}
+	new := make(map[string]interface{})
+	for k, v := range ng.Labels {
+		new[k] = v
+	}
+	return new, nil
+}
+
+func (a *mqlAwsEksNodegroup) nodeRole() (*mqlAwsIamRole, error) {
```
Comment (tas50) [inline providers/aws/resources/aws_eks.go:292]: This one is failing as well

```
  nodeRole: 1 error occurred:
	* cannot convert primitive with NO type information
```

**Comment by tas50** [comment]
LGTM from the product side

**Review by imilchev — APPROVED** [review]

**Pull request merged**